src/Controllers/MagicLinkController.php

```diff
@@ -92,7 +92,7 @@ public function loginAction()
         $user  = $this->provider->findByCredentials(['email' => $email]);
 
         if ($user === null) {
-            return redirect()->route('magic-link')->with('error', lang('Auth.invalidEmail'));
+            return redirect()->route('magic-link')->with('error', lang('Auth.invalidEmail', [$email]));
```

Comment:
The session variable `error` is a HTML string, because it is not escaped.
So there is a room to XSS attacks with email like `"><svg/onload=confirm(1)>"@gmail.com`.
https://github.com/codeigniter4/shield/blob/a38b3ccc5d70862ffc27b470a76cb4156f4d1a70/src/Views/magic_link_form.php#L13

Reply:
Is it better that we escape all session `error` `errors` in views?

Reply:
agree! it should be escaped.

Reply:
should I update it too or do I need to make a new PRs?

Reply:
Escape in the magic link view in this PR, because this PR makes output with email address.

Please send another PR for other views.

Reply:
ahh, okay.